psd-web/app/controllers/users/passwords_controller.rb

```diff
@@ -6,33 +6,14 @@ class PasswordsController < Devise::PasswordsController
                        :has_viewed_introduction,
                        only: %i(edit sign_out_before_resetting_password)
 
-    def edit
-      return render :invalid_link, status: :not_found if reset_token_invalid?
-      return render :signed_in_as_another_user, locals: { reset_password_token: params[:reset_password_token] } if wrong_user?
-      return render :expired, status: :gone if reset_token_expired?
+    skip_before_action :require_secondary_authentication
+    before_action :secondary_authentication_before_edit, only: :edit
+    before_action :secondary_authentication_before_update, only: :update
```

Comment:
Suggestion: I don't see why`require_secondary_authentication_before_update` has any reason to exist as adds 0 value over this call:
```suggestion
    before_action :require_secondary_authentication, only: :update
```